pkg/providers/pricing/pricing.go

```diff
@@ -89,23 +87,27 @@ func newZonalPricing(defaultPrice float64) zonal {
 }
 
 // NewPricingAPI returns a pricing API configured based on a particular region
-func NewAPI(sess *session.Session, region string) pricingiface.PricingAPI {
-	if sess == nil {
+func NewAPI(cfg aws.Config) *pricing.Client {
+	if cfg.Region == "" {
```

Comment:
What happens if we return `nil` here? Do we crash?

Reply:
Now that I am looking at it, would this condition ever return nil? every place it is used there is a check for region or it is specifically set right before this is called.